File: app/src/androidTest/java/co/edu/uniandes/miswmobile/vinilosapp/PerformerListTest.java
package co.edu.uniandes.miswmobile.vinilosapp;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static androidx.test.espresso.matcher.ViewMatchers.withText;
import static org.hamcrest.Matchers.allOf;

import static kotlinx.coroutines.DelayKt.delay;

import androidx.test.espresso.ViewInteraction;
import androidx.test.ext.junit.rules.ActivityScenarioRule;
import androidx.test.ext.junit.runners.AndroidJUnit4;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;
import co.edu.uniandes.miswmobile.vinilosapp.ui.AccessActivity;

@RunWith(AndroidJUnit4.class)
public class PerformerListTest {

    @Rule
    public ActivityScenarioRule<AccessActivity> mActivityTestRule = new ActivityScenarioRule<>(AccessActivity.class);

    @Test
    public void mainActivityArtistsEmptyTest() {
        ViewInteraction buttonVisitor = onView(allOf(withId(R.id.buttonVisitor), withText("Visitante"), isDisplayed()));
        buttonVisitor.perform(click());

        ViewInteraction buttonArtists = onView(allOf(withId(R.id.buttonArtists), withText("Artistas"), isDisplayed()));
        buttonArtists.perform(click());

        onView(allOf(withId(R.id.action_bar), withText("Artistas"), isDisplayed()));

        onView(allOf(withId(R.id.artistsRv), isDisplayed()));

        try {
            Thread.sleep(3000);
        } catch (InterruptedException e) {
            e.printStackTrace();
        }

    }
}

File: app/src/androidTest/java/co/edu/uniandes/miswmobile/vinilosapp/PerformerListTest.kt
package co.edu.uniandes.miswmobile.vinilosapp

import androidx.test.espresso.Espresso
import androidx.test.espresso.action.ViewActions
import androidx.test.espresso.matcher.ViewMatchers
import androidx.test.ext.junit.rules.ActivityScenarioRule
import androidx.test.ext.junit.runners.AndroidJUnit4
import co.edu.uniandes.miswmobile.vinilosapp.ui.AccessActivity
import org.hamcrest.Matchers
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class PerformerListTest {

    @Rule
    @JvmField
    var mActivityTestRule = ActivityScenarioRule(
        AccessActivity::class.java
    )

    @Test
    fun mainActivityArtistsEmptyTest() {
        val buttonVisitor = Espresso.onView(
            Matchers.allOf(
                ViewMatchers.withId(R.id.buttonVisitor),
                ViewMatchers.isCompletelyDisplayed()
            )
        )
        buttonVisitor.perform(ViewActions.click())
        val buttonArtists = Espresso.onView(
            Matchers.allOf(
                ViewMatchers.withId(R.id.buttonArtists),
                ViewMatchers.isCompletelyDisplayed()
            )
        )
        buttonArtists.perform(ViewActions.click())
        Espresso.onView(
            Matchers.allOf(
                ViewMatchers.withId(R.id.action_bar),
                ViewMatchers.withText("Artistas"),
                ViewMatchers.isCompletelyDisplayed()
            )
        )
        Espresso.onView(
            Matchers.allOf(
                ViewMatchers.withId(R.id.artistsRv),
                ViewMatchers.isCompletelyDisplayed()
            )
        )
        try {
            Thread.sleep(3000)
        } catch (e: InterruptedException) {
            e.printStackTrace()
        }
    }
}